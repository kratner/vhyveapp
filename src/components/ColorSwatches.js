import React from "react";

const ColorSwatches = props => {
  let flexDirection = "";
  let borderRadius = "";
  let margins = "";

  switch (props.layout) {
    case "horizontal":
      flexDirection = "row";
      margins = "0 " + props.spacing;
      break;
    case "vertical":
      flexDirection = "column";
      margins = props.spacing + " 0";
      break;
    default:
  }
  switch (props.shape) {
    case "square":
      borderRadius = "0";
      break;
    case "circle":
      borderRadius = "50%";
      break;
    default:
  }
  let swatches = props.colors.map((_color, index) => {
    return (
      <div
        key={index}
        style={{
          backgroundColor: _color,
          width: props.size,
          height: props.size,
          borderRadius: borderRadius,
          margin: margins,
          cursor: "pointer"
        }}
        onClick={() => props.onClick(_color)}
        onMouseOver={e => {
          e.currentTarget.style.border = "1px solid rgb(255, 255, 255)";
        }}
        onMouseOut={e => {
          e.currentTarget.style.border = "1px solid rgb(255, 255, 255, 0)";
        }}
      ></div>
    );
  });
  return (
    <div style={{ display: "flex", flexDirection: flexDirection }}>
      {swatches}
    </div>
  );
};

export default ColorSwatches;
