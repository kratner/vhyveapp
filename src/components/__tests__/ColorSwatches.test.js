import React from "react";
import { shallow } from "enzyme";
import ColorSwatches from "../ColorSwatches";

describe("ColorSwatches", () => {
  it("renders correctly", () => {
    const wrapper = shallow(
      <ColorSwatches
        layout="vertical"
        shape="circle"
        size="1em"
        colors={["red", "green", "blue"]}
        spacing=".25em"
        /*
        onSwatchClick={_color => {
          this.handleColorSwatchClick(this, _color);
        }}
        */
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
