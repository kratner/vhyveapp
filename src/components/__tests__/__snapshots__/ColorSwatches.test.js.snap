// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorSwatches renders correctly 1`] = `ShallowWrapper {}`;
